### PR TITLE
[5.2] Make $this->validate() also accept array

### PR DESCRIPTION
**Pull request by mul14**
Resend PR #10773 for Laravel 5.2.

### CODE REVIEW

File: src/Illuminate/Foundation/Validation/ValidatesRequests.php [hunk 1]
```diff
@@ -28,9 +28,13 @@
      *
      * @throws \Illuminate\Foundation\Validation\ValidationException
      */
-    public function validate(Request $request, array $rules, array $messages = [], array $customAttributes = [])
+    public function validate($request, array $rules, array $messages = [], array $customAttributes = [])
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Validation/ValidatesRequests.php]: maybe `$request` is the wrong name for this variable?

Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Validation/ValidatesRequests.php]: given this is the validate request trait, i still don't think we should allow anything other than a request in here

Comment (mul14) [inline src/Illuminate/Foundation/Validation/ValidatesRequests.php]: How about `$data`? That's the variable name used in `Validator::make($data)` https://github.com/laravel/framework/blob/5.1/src/Illuminate/Validation/Factory.php#L91

**Comment by JosephSilber** [comment]
`throwValidationException` needs the actual request object.

**Comment by GrahamCampbell** [comment]
Good point @JosephSilber.

**Pull request closed without merging**